Source/Core/VideoBackends/D3D12/D3DUtil.cpp

```diff
@@ -0,0 +1,882 @@
+// Copyright 2010 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <cctype>
+#include <list>
+#include <string>
+
+#include "VideoBackends/D3D12/D3DBase.h"
+#include "VideoBackends/D3D12/D3DCommandListManager.h"
+#include "VideoBackends/D3D12/D3DDescriptorHeapManager.h"
+#include "VideoBackends/D3D12/D3DShader.h"
+#include "VideoBackends/D3D12/D3DState.h"
+#include "VideoBackends/D3D12/D3DTexture.h"
+#include "VideoBackends/D3D12/D3DUtil.h"
+#include "VideoBackends/D3D12/GeometryShaderCache.h"
+#include "VideoBackends/D3D12/PixelShaderCache.h"
+#include "VideoBackends/D3D12/VertexShaderCache.h"
+#include "VideoBackends/D3D12/Render.h"
+
+namespace DX12
+{
+
+extern D3D12_BLEND_DESC clearblendstates12[4];
+extern D3D12_DEPTH_STENCIL_DESC cleardepthstates12[3];
+extern D3D12_BLEND_DESC resetblendstate12;
+extern D3D12_DEPTH_STENCIL_DESC resetdepthstate12;
+extern D3D12_RASTERIZER_DESC resetraststate12;
+
+namespace D3D
+{
+	inline void ResourceBarrier(ID3D12GraphicsCommandList* commandList, ID3D12Resource* resource, D3D12_RESOURCE_STATES stateBefore, D3D12_RESOURCE_STATES stateAfter, UINT subresource)
```

Comment:
Don't use `inline` for things that are used in translation-unit-local scope only